topics/materials-science/tutorials/muon-stopping-sites-muairss-uep/tutorial.md

```diff
@@ -0,0 +1,220 @@
+>    - {% icon param-file %} *"Muonated structures (.zip)"*: `Muonated Si-out.cell using Si-muairss-uep.yaml` (output of **Generate muonated structures (pm-muairss)** {% icon tool %})
+>    - {% icon param-file %} *"Charge density file (.den_fmt)"*: `Si.den_fmt` (Input dataset)
+>    - {% icon param-file %} *"CASTEP log (.castep)"*: `Si.castep` (Input dataset)
+> 2. Should generate the following output(s):
+>    - {% icon param-file %} `UEP results for Muonated Si-out.cell using Si-muairss-uep.yaml` (Output dataset)
+>
+{: .hands_on}
+
+The output of this tool is a zipped folder containing the optimised structures.
+We can then use this with the next tool in our workflow.
+
+## Run clustering for optimised structures (pm-muairss read)
+
+Finally, we can cluster the optimised positions in order to generate a short
+list of potential stopping sites.
+
+> <hands-on-title>Task description</hands-on-title>
+>
+> 1. {% tool [PyMuonSuite AIRSS Cluster](toolshed.g2.bx.psu.edu/repos/muon-spectroscopy-computational-project/pm_muairss_read/pm_muairss_read/0.2.1+galaxy1) %} with the following parameters:
+>    - {% icon param-file %} *"optimised muonated structures (.zip)"*: `UEP results for Muonated Si-out.cell using Si-muairss-uep.yaml` (output of **Run UEP optimisation (pm-uep-opt)** {% icon tool %})
+> 2. Should generate the following output(s):
+>    - {% icon param-file %} `Cluster Report` (Output dataset)
+>    - {% icon param-file %} `Cluster Data` (Output dataset)
+>
+{: .hands_on}
+
+The report will contain a human readable summary of the clustering, with the
+underlying raw data also provided. While we may have generated 10s of random
+positions, we should now see only a few potential stopping sites as during the
+optimisation stage the muons migrate to minimise the forces acting on them.
+These new positions are likely to overlap, corresponding to local energy minima.
+In addition to listing the number of muons that arrived at each stopping site,
+we also report the energy corresponding to that site. This can help inform which
+site(s) are most favourable for the muon.
+
+# Conclusion
+
+![Graphical illustration of the generation, optimisation and clustering of potential stopping sites](../../images/muon_stopping_sites_diagram.png)
+
+We have used tools from PyMuonSuite to determine possible locations of implanted
+muons in a simple structure. In reality, the structure of interest will likely
+be more complex, but the workflow will remain the same. It is also possible to
+modify the settings used by PyMuonSuite with
+{% tool [PyMuonSuite AIRSS Configure](toolshed.g2.bx.psu.edu/repos/muon-spectroscopy-computational-project/pm_yaml_config/pm_yaml_config/0.2.1+galaxy0) %}
+and then use that file in our input, or alternatively modify the workflow to
+include this tool directly. The Galaxy implementation has explanations of the
+various settings, and sensible values set by default.
```

Comment:
I think it's probably worth including configuration in the tutorial, but maybe that's a big enough task that it should be done separately.

Reply:
> the "right" way to do it would probably to modify the underlying workflow (which doesn't include config) and do it all in one?

Yes.

> Probably worth noting there is that issue considering whether we should just remove the config tool entirely https://github.com/muon-spectroscopy-computational-project/muon-galaxy-tools/issues/99

And this is the reason we should not do it now :) I just think it should get into the tutorial somehow - if that ends up being because we move the config options into pm-muairss-write, that's fine

Reply:
@joelvdavies This is the discussion that lead to replacing the config file input with one generated by the tool (essentially that it's part of the overall workflow, and allows a natural introduction of the meanings of the various parameters).